infra/cifuzz/cifuzz.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module used by CI tools in order to interact with fuzzers.
+This module helps CI tools do the following:
+  1. Build fuzzers.
+  2. Run fuzzers.
+Eventually it will be used to help CI tools determine which fuzzers to run.
+"""
+
+import argparse
+import enum
+import os
+import shutil
+import sys
+
+import fuzz_target
+
+# pylint: disable=wrong-import-position
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import build_specified_commit
+import helper
+import repo_manager
+import utils
+
+
+class Status(enum.Enum):
+  """An Enum to store the possible return codes of the cifuzz module."""
+  SUCCESS = 0
+  ERROR = 1
+  BUG_FOUND = 2
+
+
+def main():
+  """Connects fuzzers with CI tools.
+
+  Returns:
+    0 on success and 1 on failure.
+  """
+  parser = argparse.ArgumentParser(
+      description='Help CI tools manage specific fuzzers.')
+
+  subparsers = parser.add_subparsers(dest='command')
+  build_fuzzer_parser = subparsers.add_parser(
+      'build_fuzzers', help='Build an OSS-Fuzz projects fuzzers.')
+  build_fuzzer_parser.add_argument('project_name')
+  build_fuzzer_parser.add_argument('github_repo_name')
+  build_fuzzer_parser.add_argument('commit_sha')
+
+  run_fuzzer_parser = subparsers.add_parser(
+      'run_fuzzers', help='Run an OSS-Fuzz projects fuzzers.')
+  run_fuzzer_parser.add_argument('project_name')
+  run_fuzzer_parser.add_argument('fuzz_time')
+  args = parser.parse_args()
+
+  # Get the shared volume directory and creates required directory.
+  if os.environ['GITHUB_WORKSPACE']:
+    git_workspace = os.path.join(os.environ['GITHUB_WORKSPACE'], 'storage')
+    if not os.path.exists(git_workspace):
+      os.mkdir(git_workspace)
+    out_dir = os.path.join(os.environ['GITHUB_WORKSPACE'], 'out')
+    if not os.path.exists(out_dir):
+      os.mkdir(out_dir)
```

Comment:
just use `os.makedirs(out_dir, exist_ok=True)` here, which will created intermediate dirs etc and avoid these exists checks.

Reply:
Done. That is a good tip.